docs/user/openstack/observability.md

```diff
@@ -0,0 +1,388 @@
+of the HTTP calls from the observer to the observed object ("pull model").
+
+In the following instructions, instead of using an external arbitrary
+Prometheus instance, we will be using RHOSO's Prometheus as the collector of
+both OpenShift and OpenStack metrics.
+
+OpenShift exposes a federation endpoint to expose a subset of metrics to an
+external scraper. You can follow [these instructions][federation] to get
+acquainted to the endpoint.
+
+[federation]: https://docs.redhat.com/en/documentation/openshift_container_platform/4.17/html/monitoring/accessing-third-party-monitoring-apis#monitoring-querying-metrics-by-using-the-federation-endpoint-for-prometheus_accessing-monitoring-apis-by-using-the-cli "OpenShift documentation: Querying metrics by using the federation endpoint for Prometheus"
+
+#### Step 1: Gather credentials and coordinates
+
+While connected to the OpenShift cluster through a username identified by password (as opposed to logging in using the `kubeconfig` file generated by the installer), fetch a token:
+
+```bash
+oc whoami -t
+```
+
+Then get the Prometheus federation route URL:
+
+```bash
+oc -n openshift-monitoring get route prometheus-k8s-federate -ojsonpath={'.status.ingress[].host'}
+```
+
+#### Let RHOSO scrape OpenShift's federation endpoint
+
+As stated in the [OpenShift documentation][ocp-federation-docs], it is recommended to limit scraping
+to fewer than 1000 samples for each request, and with a maximum frequency of
+once every 30 seconds.
+
+[ocp-federation-docs]: https://docs.openshift.com/container-platform/4.17/observability/monitoring/accessing-third-party-monitoring-apis.html#monitoring-querying-metrics-by-using-the-federation-endpoint-for-prometheus_accessing-monitoring-apis-by-using-the-cli
+
+In this example, we will only request two metrics: `kube_node_info` and
+`kube_persistentvolume_info` (see the `params.match[]` query below).
+
+While connected to the RHOSO cluster, apply this manifest:
+
+```yaml
+apiVersion: monitoring.rhobs/v1alpha1
+kind: ScrapeConfig
+metadata:
+  labels:
+    service: metricStorage
+  name: sos1-federated
+  namespace: openstack
+spec:
+  params:
+    'match[]':
+    - '{__name__=~"kube_node_info|kube_persistentvolume_info"}'
+  metricsPath: '/federate'
+  authorization:
+    type: Bearer
+    credentials:
+      name: ocp-federated
+      key: token
+  scheme: HTTPS # or HTTP
+  scrapeInterval: 30s
+  staticConfigs:
+  - targets:
+    - prometheus-k8s-federate-openshift-monitoring.apps.openshift.example # This is the URL fetched previously
+  # add a tlsConfig stanza in case the endpoint is HTTPS but uses a custom CA
+```
+
+Don't forget to make the token available as a secret (in the example above, the name is `ocp-federated`):
+
+```bash
+oc -n openstack ceate secret generic ocp-federated --from-literal=token=<the token fetched previously>
+```
+
+Once the new scrapeconfig propagates, the requested OpenShift metrics will be
+accessible for querying in RHOSO's OpenShift UI.
```

Comment:
Is "OpenShift UI" on purpose or should it be "Prometheus UI". I'm not familiar enough yet as I'm still configuring RHOSO to get prometheus running.

Reply:
I think it was correct after seeing the RHOSO telemetry depends on OpenShift observability framework.